src/source/linear_ramp.rs

```diff
@@ -0,0 +1,124 @@
+use std::time::Duration;
+
+use crate::{Sample, Source};
+
+/// Internal function that builds a `LinearRamp` object.
+pub fn linear_gain_ramp<I>(
+    input: I,
+    duration: Duration,
+    start_gain: f32,
+    end_gain: f32,
+) -> LinearGainRamp<I>
+where
+    I: Source,
+    I::Item: Sample,
+{
+    let duration = duration.as_secs() * 1000000000 + duration.subsec_nanos() as u64;
```

Comment:
is there an advantage of this vs: https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_secs_f32?

maybe this pr is from before that function was added :sweat_smile: